doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -105,197 +129,222 @@ simply :g:`t=u` dropping the implicit type of :g:`t` and :g:`u`.
+     entire hypotheses and/or the goal; it must not include any
+     :n:`at @occs_nums` clauses.
 
    .. exn:: Not convertible.
       :undocumented:
 
-   .. tacv:: change @term with @term’
+   .. exn:: Too few occurrences.
+      :undocumented:
 
-      This replaces the occurrences of :n:`@term` by :n:`@term’` in the current goal.
-      The term :n:`@term` and :n:`@term’` must be convertible.
+   .. tacn:: now_show @one_term
 
-   .. tacv:: change @term at {+ @natural} with @term’
+      This is a synonym for :n:`change @conversion`. It can be used to
+      make some proof steps explicit when refactoring a proof script
+      to make it readable.
 
-      This replaces the occurrences numbered :n:`{+ @natural}` of :n:`@term` by :n:`@term’`
-      in the current goal. The terms :n:`@term` and :n:`@term’` must be convertible.
+   .. seealso:: :ref:`Performing computations <performingcomputations>`
 
-      .. exn:: Too few occurrences.
-         :undocumented:
+.. tacn:: change_no_check {? @one_term__from {? at @occs_nums } with } @one_term__to {? @occurrences }
 
-   .. tacv:: change @term {? {? at {+ @natural}} with @term} in @goal_occurrences
+   For advanced usage. Similar to :tacn:`change` :n:`@term`, but as an optimization,
+   it skips checking that :n:`@one_term__to` is convertible with the goal or
+   :n:`@one_term__from`.
 
-      In the presence of :n:`with`, this applies :tacn:`change` to the
-      occurrences specified by :n:`@goal_occurrences`. In the
-      absence of :n:`with`, :n:`@goal_occurrences` is expected to
-      only list hypotheses (and optionally the conclusion) without
-      specifying occurrences (i.e. no :n:`at` clause).
+   Recall that the Coq kernel typechecks proofs again when they are concluded to
+   ensure correctness. Hence, using :tacn:`change` checks convertibility twice
+   overall, while :tacn:`change_no_check` can produce ill-typed terms,
+   but checks convertibility only once.
+   Hence, :tacn:`change_no_check` can be useful to speed up certain proof
+   scripts, especially if one knows by construction that the argument is
+   indeed convertible to the goal.
 
-   .. tacv:: now_show @term
+   In the following example, :tacn:`change_no_check` replaces :g:`False` with
+   :g:`True`, but :cmd:`Qed` then rejects the proof, ensuring consistency.
 
-      This is a synonym of :n:`change @term`. It can be used to
-      make some proof steps explicit when refactoring a proof script
-      to make it readable.
+   .. example::
 
-   .. seealso:: :ref:`Performing computations <performingcomputations>`
+      .. coqtop:: all abort
+
+         Goal False.
+           change_no_check True.
+           exact I.
+         Fail Qed.
+
+   .. example::
+
+      .. coqtop:: all abort
+
+         Goal True -> False.
+           intro H.
+           change_no_check False in H.
+           exact H.
+         Fail Qed.
+
+   .. tacn:: convert_concl_no_check @one_term
+
+      .. todo PR: Can we remove this now?
```

Comment:
Yes, we should be able to remove it in 8.14. However, since this PR is targeting 8.13, we should keep the documentation until the removal. Thus, it would be better to wait for this PR to be merged before opening the PR to remove it.